cli/user.go

```diff
@@ -121,18 +121,48 @@ func runSetUser(cmd *cobra.Command, args []string) {
 		mustUsage(cmd)
 		return
 	}
-	hashed, err := security.PromptForPasswordAndHash()
-	if err != nil {
-		log.Error(err)
-		return
+	var err error
+	var hashed []byte
+	switch password {
+	case "":
+		hashed, err = security.PromptForPasswordAndHash()
+		if err != nil {
+			panic(err)
+		}
+	case "-":
+		scanner := bufio.NewScanner(os.Stdin)
+		if scanner.Scan() == true {
```

Comment:
`== true` is a silly pattern, `scanner.Text()` allocates, and there's no need to `util.Erorrf` before calling `panic()`. How about:
```go
	scanner := bufio.NewScanner(os.Stdin)
	if scanner.Scan() {
		if b := scanner.Bytes(); len(b) > 0 {
			hashed, err = security.HashPassword(b)
			if err != nil {
				panic(err)
			}
			if scanner.Scan() {
				panic("multiline passwords are not permitted")
			}
			if err := scanner.Err(); err != nil {
				panic(err)
			}

			break // Success.
		}
	} else {
		if err := scanner.Err(); err != nil {
			panic(err)
		}
	}

	panic("empty passwords are not permitted")
```